tests/test_allauth_2fa.py

```diff
@@ -60,3 +60,27 @@ def test_invalid_2fa_login(self):
         resp = self.client.post(reverse('two-factor-authenticate'),
                                 {'otp_token': 'invalid'})
         self.assertEqual(resp.status_code, 200)
+
+    def test_2fa_reset_flow(self):
+        """Test login flow reset when navigating away."""
+        user = get_user_model().objects.create(username='john')
+        user.set_password('doe')
+        user.save()
+        user.totpdevice_set.create()
+
+        resp = self.client.post(reverse('account_login'),
+                                {'login': 'john',
+                                 'password': 'doe'})
+        self.assertRedirects(resp,
+                             reverse('two-factor-authenticate'),
+                             fetch_redirect_response=False)
+
+        # Navigate back to login
+        self.client.get(reverse('account_login'))
```

Comment:
Can we assert that `allauth_2fa_user_id` is in the session before this call and not after this call?